src/main/java/com/nerodesk/App.java

```diff
@@ -100,113 +106,132 @@ public App(final Base base) throws IOException {
+                                base.user(
+                                    new RqAuth(req).identity().urn()
+                                ).docs().doc(file)
+                            );
                         }
-                    )
-                )
-            ),
-            new FkRegex(
-                "/r",
-                new Takes() {
-                    @Override
-                    public Take route(final Request req) throws IOException {
-                        final String file =
-                            new RqHref(req).href().param("f").iterator().next();
-                        return new TkRead(
-                            base.user(
-                                new RqAuth(req).identity().urn()
-                            ).docs().doc(file)
-                        );
                     }
-                }
-            ),
-            new FkRegex(
-                "/d",
-                new Takes() {
-                    @Override
-                    public Take route(final Request req) throws IOException {
-                        final String file =
-                            new RqHref(req).href().param("x").iterator().next();
-                        return new TkDelete(
-                            base.user(
-                                new RqAuth(req).identity().urn()
-                            ).docs().doc(file)
-                        );
+                ),
+                new FkRegex(
+                    "/w",
+                    new Takes() {
+                        @Override
+                        public Take route(final Request req)
+                            throws IOException {
+                            return new TkWrite(
+                                base.user(
+                                    new RqAuth(req).identity().urn()
+                                ).docs(),
+                                req
+                            );
+                        }
                     }
-                }
+                )
             ),
-            new FkRegex(
-                "/w",
-                new Takes() {
-                    @Override
-                    public Take route(final Request req) throws IOException {
-                        return new TkWrite(
-                            base.user(
-                                new RqAuth(req).identity().urn()
-                            ).docs(),
-                            req
-                        );
-                    }
+            new Fallback() {
+                @Override
+                public Take take(final RqFallback req) {
+                    final String exc = ExceptionUtils.getStackTrace(
+                        req.throwable()
+                    );
+                    Logger.info(this, "Exception thrown\n%s", exc);
+                    return new TkHTML(
+                        String.format("oops, something went wrong!\n%s", exc)
```

Comment:
well, it should be formatted as HTML if you're using `TkHTML`... I thought you would something like rultor is doing, with an HTML template that renders the stacktrace.